src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentNotebookRequest.cs

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /// <summary>
+    /// SQL Agent create Notebook params
+    /// </summary>
+    public class CreateAgentNotebookParams : TaskRequestDetails
+    {
+        public string OwnerUri { get; set; }
+        public AgentNotebookInfo Notebook { get; set; }
+        public string TemplateFilePath { get; set; }
+    }
+
+    /// <summary>
+    /// SQL Agent create Notebook result
+    /// </summary>
+    public class CreateAgentNotebookResult : ResultStatus
+    {
+        public AgentNotebookInfo Job { get; set; }
+    }
+
+    /// <summary>
+    /// SQL Agent create Notebook request type
+    /// </summary>
+    public class CreateAgentNotebookRequest
+    {
+        public static readonly
+            RequestType<CreateAgentNotebookParams, CreateAgentNotebookResult> Type =
+            RequestType<CreateAgentNotebookParams, CreateAgentNotebookResult>.Create("agent/createnotebook");
+    }
+
+    /// <summary>
+    /// SQL Agent update Notebook params
+    /// </summary>
+    public class UpdateAgentNotebookParams : TaskRequestDetails
+    {
+        public string OwnerUri { get; set; }
+
+        public string OriginalNotebookName { get; set; }
+
+        public AgentNotebookInfo Notebook { get; set; }
+        public string TemplateFilePath { get; set; }
+
+    }
+
+    /// <summary>
+    /// SQL Agent update Notebook result
+    /// </summary>
+    public class UpdateAgentNotebookResult : ResultStatus
+    {
+    }
+
+    /// <summary>
+    /// SQL Agent update Notebook request type
+    /// </summary>
+    public class UpdateAgentNotebookRequest
+    {
+        /// <summary>
+        /// Request definition
+        /// </summary>
+        public static readonly
+            RequestType<UpdateAgentNotebookParams, UpdateAgentNotebookResult> Type =
+            RequestType<UpdateAgentNotebookParams, UpdateAgentNotebookResult>.Create("agent/updatenotebook");
+    }
+
+    /// <summary>
+    /// SQL Agent delete Notebook params
+    /// </summary>
+    public class DeleteAgentNotebookParams : TaskRequestDetails
+    {
+        public string OwnerUri { get; set; }
+
+        public AgentNotebookInfo Notebook { get; set; }
+    }
+
+    /// <summary>
+    /// SQL Agent delete Notebook request type
+    /// </summary>
+    public class DeleteAgentNotebookRequest
+    {
+        /// <summary>
+        /// Request definition
+        /// </summary>
+        public static readonly
+            RequestType<DeleteAgentNotebookParams, ResultStatus> Type =
+            RequestType<DeleteAgentNotebookParams, ResultStatus>.Create("agent/deletenotebook");
+    }
+
+    /// <summary>
+    /// SQL Agent Notebook materialized params
+    /// </summary>
+    public class AgentNotebookMaterializedParams : TaskRequestDetails
+    {
+        public string OwnerUri { get; set; }
+        public string TargetDatabase { get; set; }
+        public int NotebookMaterializedID { get; set; }
```

Comment:
in C# the convention is abbreviations should be Pascal-cased.  i.e. so `Id` here.